packages/@aws-cdk/aws-iot-actions/lib/s3-put-object-action.ts

```diff
@@ -0,0 +1,71 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as iot from '@aws-cdk/aws-iot';
+import * as s3 from '@aws-cdk/aws-s3';
+import { kebab as toKebabCase } from 'case';
+import { CommonActionProps } from './private/common-action-props';
+import { singletonActionRole } from './private/role';
+
+/**
+ * Configuration properties of an action for s3.
+ */
+export interface S3PutObjectActionProps extends CommonActionProps {
+  /**
+   * The Amazon S3 canned ACL that controls access to the object identified by the object key.
+   * @see https://docs.aws.amazon.com/AmazonS3/latest/userguide/acl-overview.html#canned-acl
+   *
+   * @default None
+   */
+  readonly cannedAcl?: s3.BucketAccessControl;
```

Comment:
I don't like the name `cannedAcl`. Let's call this `accessControl`, like we do in CodePipeline's `S3DeployAction`.